File: 2020/day01/rust/src/part1.rs
use std::collections::HashSet;

pub fn parse(lines: &Vec<&str>) -> Vec<i32> {
    let mut values: Vec<i32> = vec![0; lines.len()];
    for index in 0..lines.len() {
        values[index] = lines[index].parse().unwrap();
    }
    return values;
}

pub fn find_pair(values: &Vec<i32>, target: i32) -> [i32; 2] {
    let mut result: [i32; 2] = [0, 0];
    let values: HashSet<_> = values.iter().cloned().collect();

    for value in &values {
        if values.contains(&(target - value)) {
            result = [*value, target - value];
            result.sort();
            break;
        }
    }

    return result;
}

pub fn compute_answer(lines: &Vec<&str>, target: i32) -> i32 {
    let values = parse(&lines);
    let results = find_pair(&values, target);
    return results[0] * results[1];
}

File: 2015/day08/python/part1.py
from typing import List


def compute_answer(lines: List[str]) -> int:
    result = 0
    for line in lines:
        index = 0
        while index < len(line):
            # print(index, line[index])
            if line[index] == '"':
                if index == 0:
                    result += 1
                elif index == len(line) - 1:
                    result += 1
            elif line[index] == "\\":
                if line[index + 1] == "\\":
                    result += 1
                    index += 1
                elif line[index + 1] == '"':
                    result += 1
                    index += 1
                elif line[index + 1] == "x":
                    result += 3
                    index += 3
            index += 1
    return result
